gitlab/utils.py

```diff
@@ -113,17 +113,17 @@ class EncodedId(str):
         https://docs.gitlab.com/ee/api/index.html#path-parameters
     """
 
-    # mypy complains if return type other than the class type. So we ignore issue.
-    def __new__(  # type: ignore
-        cls, value: Union[str, int, "EncodedId"]
-    ) -> Union[int, "EncodedId"]:
-        if isinstance(value, (int, EncodedId)):
+    def __new__(cls, value: Union[str, int, "EncodedId"]) -> "EncodedId":
+        if isinstance(value, EncodedId):
             return value
 
-        if not isinstance(value, str):
+        if isinstance(value, int):
+            return_value = str(value)
+        elif isinstance(value, str):
+            return_value = urllib.parse.quote(value, safe="")
+        else:
             raise TypeError(f"Unsupported type received: {type(value)}")
-        value = urllib.parse.quote(value, safe="")
-        return super().__new__(cls, value)
+        return super().__new__(cls, return_value)
```

Comment:
Would you be against just redefining `value` here? Then all returns are the same, seems less visual load for me :)

Reply:
Mypy will complain if that is done. As redefining the type of the int

Reply:
While driving around I realized that the `str` `__new__()` method will happily take an `int` value. So it simplified the code.

Reply:
Nice!